rn/Teacher/ios/Teacher/SpeedGrader/RubricAssessor.swift

```diff
@@ -0,0 +1,207 @@
+                RubricCriteriaAssessor(criteria: criteria)
+            }
+        }
+            .multilineTextAlignment(.leading)
+            .padding(.horizontal, 16)
+    }
+
+    func RubricCriteriaAssessor(criteria: Rubric) -> some View { VStack(alignment: .leading, spacing: 0) {
+        let assessment = assessments[criteria.id] ?? submission.rubricAssessments?[criteria.id].map {
+            APIRubricAssessment(comments: $0.comments, points: $0.points, rating_id: $0.ratingID)
+        }
+
+        HStack { Spacer() }
+        Text(criteria.desc)
+            .font(.semibold16).foregroundColor(.textDarkest)
+        if criteria.ignoreForScoring {
+            Text("This criterion will not impact the score.")
+                .font(.regular12).foregroundColor(.textDark)
+                .padding(.top, 2)
+        }
+
+        FlowStack(spacing: UIOffset(horizontal: 8, vertical: 8)) { leading, top in
+            if !assignment.freeFormCriterionCommentsOnRubric, let ratings = criteria.ratings {
+                ForEach(ratings.reversed(), id: \.id) { rating in
+                    let isSelected = assessment?.rating_id == rating.id
+                    CircleToggle(isOn: Binding(get: { isSelected }, set: { newValue in
+                        assessments[criteria.id] = newValue ? APIRubricAssessment(
+                            comments: assessment?.comments,
+                            points: rating.points,
+                            rating_id: rating.id
+                        ) : APIRubricAssessment(comments: assessment?.comments)
+                    })) {
+                        Text((isSelected ? assessment?.points : nil) ?? rating.points)
+                    }
+                        .alignmentGuide(.leading, computeValue: leading)
+                        .alignmentGuide(.top, computeValue: top)
+                }
+            }
+
+            let customGrade = (assignment.freeFormCriterionCommentsOnRubric || assessment?.rating_id == nil)
+                ? assessment?.points : nil
+            CircleToggle(isOn: Binding(get: { customGrade != nil }, set: { newValue in
+                if newValue {
+                    promptCustomGrade(criteria, assessment: assessment)
+                } else {
+                    assessments[criteria.id] = APIRubricAssessment(comments: assessment?.comments)
+                }
+            })) {
+                if let grade = customGrade {
+                    Text(grade)
+                } else {
+                    Icon.addSolid
+                }
+            }
+                .accessibility(label: Text("Customize Grade"))
+                .alignmentGuide(.leading, computeValue: leading)
+                .alignmentGuide(.top, computeValue: top)
+        }
+            .padding(.top, 8)
+
+        let showAdd = assignment.freeFormCriterionCommentsOnRubric && assessment?.comments?.isEmpty != false
+        let showLong = criteria.longDesc.isEmpty == false
```

Comment:
I saw this the other day: https://forums.swift.org/t/boolean-comparison-causes-extremely-slow-compilation/41609

Apparently `foo == false` is quite a bit slower than `!foo` which is sad because i think the former is more expressive.

Not looking for a change but I thought it was interesting and thought you might too.

Reply:
We use the `==` mostly when dealing with `Optional<Bool>`, but that doesn't look to be the case here.